packages/grid/x-data-grid/src/components/menu/GridMenu.tsx

```diff
@@ -70,6 +70,17 @@ function GridMenu(props: GridMenuProps) {
   const rootProps = useGridRootProps();
   const classes = useUtilityClasses(rootProps);
 
+  const savedFocusRef = React.useRef<HTMLElement | null>(null);
+  React.useEffect(() => {
+    if (open) {
+      savedFocusRef.current =
+        document.activeElement instanceof HTMLElement ? document.activeElement : null;
+    } else {
+      savedFocusRef.current?.focus?.();
+      savedFocusRef.current = null;
+    }
+  }, [open]);
+
```

Comment:
I have very mixed feelings about this solution. It's small and it works, but it relies on the fact that:
 - we know the focus is going to be managed downstream when this prop changes
 - this effect runs before the effect downstream (before the focus is programatically modified), which I'm not sure if we can guarantee

If you look at the previous commit, there is also another somewhat sacrilegious implementation that does the same thing but during the render phase :| which ensures we save the focus before it has a chance to be modified, but it's a side-effect in the render phase.

Please let me know if you think this is acceptable, otherwise we can look into other solutions.

Reply:
Looks good to me.
What do you think about using `useLayoutEffect` (or rather `useEnhancedEffect`) instead?